File: tests/java/tests/java/org/elastos/hive/config/NodeConfig.java
package org.elastos.hive.config;

import com.fasterxml.jackson.annotation.JsonProperty;

public class NodeConfig extends Config {
	@JsonProperty("provider")
	private String provider;
	// the service instance did of backup node.
	@JsonProperty("targetDid")
	private String targetDid;
	// the host of backup node.
	@JsonProperty("targetHost")
	private String targetHost;
	@JsonProperty("storePath")
	private String storePath;

	public String provider() {
		return this.provider;
	}

	public String targetDid() {
		return this.targetDid;
	}

	public String targetHost() {
		return this.targetHost;
	}

	public String storePath() {
		return this.storePath;
	}

	public static NodeConfig deserialize(String content) {
		return deserialize(content, NodeConfig.class);
	}
}

File: src/config/nodeconfig.ts
export class NodeConfig {
	private provider: string;
	// the service instance did of backup node.
	private targetDid: string;
	// the host of backup node.
	private targetHost: string
	private storePath: string;

	public getProvider(): string {
		return this.provider;
	}

	public getTargetDid(): string {
		return this.targetDid;
	}

	public getTargetHost(): string {
		return this.targetHost;
	}

	public getStorePath(): string {
		return this.storePath;
	}

	public static deserialize(json: any): NodeConfig {
		let jsonObj = (typeof json === "string" ? JSON.parse(json as string) : json);
		let newConfig = new NodeConfig();

		newConfig.provider = jsonObj['provider'];
		newConfig.targetDid = jsonObj['targetDid'];
		newConfig.targetHost = jsonObj['targetHost'];
		newConfig.storePath = jsonObj['storePath'];

		return newConfig;
	}
}
